crt_portal/cts_forms/views.py

```diff
@@ -635,10 +640,52 @@ def post(self, request):
             return render(request, 'forms/complaint_view/actions/index.html', output)
 
 
-class SaveReportAttachmentView(LoginRequiredMixin, FormView):
+class ReportAttachmentView(LoginRequiredMixin, FormView):
     """Can be used for saving attachments for a report"""
     form_class = AttachmentActions
-    http_method_names = ['post']
+    http_method_names = ['get', 'post']
+
+    def get(self, request, id, filename):
+        """
+        Download a particular attachment for a report
+        """
+        report = get_object_or_404(Report.objects.prefetch_related('attachments'), pk=id)
+        attachment = report.attachments.get(file=f'attachments/{filename}')
+
+        logger.info(f'User {request.user} downloading attachment {attachment.filename} ({attachment.file.name}) for report {id}')
+
+        if settings.ENABLE_LOCAL_ATTACHMENT_STORAGE:
+            try:
+                file = open(attachment.file.name, 'rb')
+                mime_type, _ = mimetypes.guess_type(attachment.filename)
+                response = HttpResponse(file, content_type=mime_type)
+                response['Content-Disposition'] = f'attachment;filename={attachment.filename}'
+                return response
+
+            except FileNotFoundError:
+                raise Http404(f'File {attachment.filename} not found.')
+
+        else:
+            # Generate a presigned URL for the S3 object
+            s3_client = boto3.client(
+                service_name='s3',
+                aws_access_key_id=settings.PRIV_S3_ACCESS_KEY_ID,
+                aws_secret_access_key=settings.PRIV_S3_SECRET_ACCESS_KEY,
+                endpoint_url=settings.PRIV_S3_ENDPOINT_URL,
+                config=Config(signature_version='s3v4'))
+
+            try:
+                response = s3_client.generate_presigned_url('get_object',
+                                                            Params={'Bucket': settings.PRIV_S3_BUCKET,
+                                                                    'Key': attachment.file.name,
+                                                                    'ResponseContentDisposition': f'attachment;filename={attachment.filename}'},
+                                                            ExpiresIn=30)
+
+                return redirect(response)
+
+            except ClientError as e:
+                logging.error(e)
```

Comment:
Would be great to send this to NewRelic in a later iteration.

Reply:
Agreed, we can explore in a different card.